### PR TITLE
Okta auth with fake user data bypass

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
Enable Okta—
But leave another way in
With parameters.

Merging this will wrap the application in Okta authentication, going through https://test.idp.idm.cms.gov/ and requiring CMS EUA credentials. Even on localhost. However, appending one of:

- `?dev=dev-ak` 
- `?dev=dev-az` 
- `?dev=dev-ma` 

to the URL will bypass this and provide access as a fake user associated with the corresponding state.

Currently CMS EUA authentication isn't working for me at all, even though it worked fine last week. That doesn't appear to be related to this PR, though.

Testing it locally before merging would probably be good…

This also adds support for the `context_data.show_if_state_program_type_in` logic at the Part level; this was basically necessary to add as a way of verifying that the different states' info was being picked up from the fake data.

Addresses #479, also moves #480 and #392 along. Probably makes #376 unnecessary.

### CODE REVIEW

File: frontend/react/src/components/layout/Header.js [hunk 1]
```diff
@@ -55,7 +56,6 @@ class Header extends Component {
                       <a href="javascript:void(0)">Manage account</a>
                     </li>
                     <li className="logout">
-                      <a href="javascript:void(0)">Log out</a>
```
Comment (mgwalker) [inline frontend/react/src/components/layout/Header.js:58]: Does this need `<Logout />` added?

File: frontend/react/src/actions/initial.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ export const QUESTION_ANSWERED = "QUESTION ANSWERED";
 const temp__data = require("./initial.json");
 
 export const loadSections = () => {
+  console.log("loadSections");
```
Comment (mgwalker) [inline frontend/react/src/actions/initial.js:9]:
```suggestion
```

**Review by mgwalker — APPROVED** [review]
I don't love it, but it seems like a good iteration towards the goal. It's hard to love some of the incremental steps. 🤷‍♂️ 

I added a lot of suggestions for removing unused code/log statements and one question, but nothing worth holding this up for. I'm going to approve it and let you decide whether to merge now or clean anything up first.

File: frontend/react/src/components/layout/Header.js (hunk 1 above)
Comment (tadhg-ohiggins) [reply frontend/react/src/components/layout/Header.js:58]: It needs that handled somehow; right now, if that component is rendered the whole page errors out if you're not authenticated via Okta. I don't think that should hold up the PR, though.

**Pull request merged**